packages/rum-core/src/state.js

```diff
@@ -28,7 +28,14 @@ const state = {
   //  Time when agent is bootstrapped and patching of modules happens
   bootstrapTime: null,
   // Time when the document is last backgrounded
-  lastHiddenStart: Number.MIN_SAFE_INTEGER
+  lastHiddenStart: Number.MIN_SAFE_INTEGER,
+  // Tracking info of cumulative layout shift calculation
+  cls: {
```

Comment:
Since the max-session-gap1s-limit5s technique requires to have into account all the layout shift events over time a state has been created to ensure the proper calculation.

This state is only accessed/modified in the CLS calculation.

I'm open to proposals in case you prefer other ways of handling this. 😊